object/slicer/slicer.go

```diff
@@ -0,0 +1,596 @@
+package slicer
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"hash"
+	"io"
+
+	"github.com/nspcc-dev/neofs-sdk-go/checksum"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/session"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+	"github.com/nspcc-dev/neofs-sdk-go/version"
+	"github.com/nspcc-dev/tzhash/tz"
+)
+
+// ObjectWriter represents a virtual object recorder.
+type ObjectWriter interface {
+	// InitDataStream initializes and returns a stream of writable data associated
+	// with the object according to its header. Provided header includes at least
+	// container, owner and object ID fields.
+	InitDataStream(header object.Object) (dataStream io.Writer, err error)
+}
+
+// Slicer converts input raw data streams into NeoFS objects. Working Slicer
+// must be constructed via New.
+type Slicer struct {
+	signer neofscrypto.Signer
+
```

Comment:
Do we need this much emptiness here?

Reply:
helps to avoid fmt's alignment, any problems with linebreaks?

Reply:
Yeah, I love when they mean something.

Reply:
I separate independent entities by them, modification of one of the code lines won't affect the other one

Reply:
I know the problem. It's not worth this much of a trouble to me. But works either way of course.